### PR TITLE
Fix autocomplete when pressing enter

### PR DESCRIPTION
**Pull request by 3v0k4**
Before this if the user filled "aa" and with the keyboard they selected
"aaBB" the app would have selected "aa".

### CODE REVIEW

File: app/javascript/angular/code/filtersUtils.js [hunk 1]
```diff
@@ -59,23 +59,15 @@ export const setupTimeRangeFilter = (callback, timeFrom, timeTo) => {
 
 export const setupAutocomplete = (callback, id, path) => {
   if (document.getElementById(id)) {
-    $("#" + id)
-      .bind("keydown", function(event) {
```
Comment (3v0k4) [inline app/javascript/angular/code/filtersUtils.js]: This was the core of the problem

File: app/javascript/elm/src/LabelsInput.elm [hunk 2]
```diff
@@ -51,11 +51,16 @@ update msg model toCmd =
             ( withCandidate str model, Cmd.none )
 
         Add str ->
-            let
-                newModel =
-                    add str model
-            in
-            ( newModel, toCmd (asList newModel) )
+            case str of
```
Comment (3v0k4) [inline app/javascript/elm/src/LabelsInput.elm]: By removing the code above the input sends empty string so we need to guard that

Comment (3v0k4) [reply app/javascript/elm/src/LabelsInput.elm]: Could not find a better way :(

File: app/javascript/angular/code/filtersUtils.js (hunk 1 above)
Comment (gitjul) [reply app/javascript/angular/code/filtersUtils.js]: cool, so the library handles that on it's own

File: app/javascript/elm/src/LabelsInput.elm (hunk 2 above)
Comment (AkkarinForest) [reply app/javascript/elm/src/LabelsInput.elm]: Yeah, I remember having that problem as well. I guess we could consider creating a custom type and having `List Label`.

File: app/javascript/angular/code/filtersUtils.js (hunk 1 above)
Comment (3v0k4) [reply app/javascript/angular/code/filtersUtils.js]: It seems so

File: app/javascript/elm/src/LabelsInput.elm (hunk 2 above)
Comment (3v0k4) [reply app/javascript/elm/src/LabelsInput.elm]: Something like a nonempty string?

**Comment by 3v0k4** [comment]
I thought about it but then a glitch in loading the autocomplete list would totally prevent users from inputting anything. Could you bring it to the call if you feel strong about that @gitjul please?

**Comment by gitjul** [comment]
yeah. I wouldn't assume it would prevent users from doing stuff :) enter is easy to press. saw the clearing behaviour in another app, I think it's better in terms of visualising applied filters (no non-existent stuff appears)

**Pull request merged**